src/mds/CInode.cc

```diff
@@ -1963,6 +1970,13 @@ void CInode::finish_scatter_update(ScatterLock *lock, CDir *dir,
       
       assert(!dir->is_frozen());
       mut->auth_pin(dir);
+
+      if (lock->get_type() == CEPH_LOCK_INEST && 
+	  !is_auth() && lock->get_next_state() == LOCK_MIX) {
```

Comment:
need locker->wrlock_force(&nestlock) here, add call drop_locks in CInode::_finish_frag_update().